### PR TITLE
[BE-4556] Support for overwrite keyword in Insert 

### PR DESCRIPTION
**Pull request by keatoooon**
All changes to the reference document were made via copy paste.

### CODE REVIEW

File: core/src/main/codegen/templates/Parser.jj [hunk 1]
```diff
@@ -324,7 +324,11 @@ void SqlSelectKeywords(List<SqlLiteral> keywords) :
 void SqlInsertKeywords(List<SqlLiteral> keywords) :
 {}
 {
-    E()
+    // I looked through the various available parsers, none of them seemed to extend this in any way,
```
Comment (jsternberg) [inline core/src/main/codegen/templates/Parser.jj]: Yea so this probably confirms that I was wrong about the being able to overwrite things. You can remove this comment since it won't be helpful for future people reading the code.

I've asked on the mailing list about this. It feels like these must be here for some reason that I just don't know how to hook into properly and I want to figure out if we're doing something wrong or if I'm just making a faulty assumption.

File: core/src/main/java/org/apache/calcite/sql/SqlInsert.java [hunk 2]
```diff
@@ -44,7 +44,8 @@ public class SqlInsert extends SqlCall {
               operands[1],
               operands[2],
               (SqlNodeList) operands[3],
-              operands[4]);
+              operands[4]
```
Comment (jsternberg) [inline core/src/main/java/org/apache/calcite/sql/SqlInsert.java]: Unnecessary formatting change.

File: core/src/main/java/org/apache/calcite/sql/SqlInsert.java [hunk 3]
```diff
@@ -99,6 +100,12 @@ public final boolean isUpsert() {
     return getModifierNode(SqlInsertKeyword.UPSERT) != null;
   }
 
+  /** Returns whether this insert contains an OVERWRITE clause.
+   */
+  public final boolean isOverWrite() {
```
Comment (jsternberg) [inline core/src/main/java/org/apache/calcite/sql/SqlInsert.java]: Nitpick: I prefer `isOverwrite`. I think of it as one word and not two.

**Review by jsternberg — CHANGES_REQUESTED** [review]
A few nitpicks but nothing too major.

File: core/src/main/codegen/templates/Parser.jj (hunk 1 above)
Comment (keatoooon) [reply core/src/main/codegen/templates/Parser.jj]: Done

File: core/src/main/java/org/apache/calcite/sql/SqlInsert.java (hunk 3 above)
Comment (keatoooon) [reply core/src/main/java/org/apache/calcite/sql/SqlInsert.java]: done

File: core/src/main/java/org/apache/calcite/sql/SqlInsert.java (hunk 2 above)
Comment (keatoooon) [reply core/src/main/java/org/apache/calcite/sql/SqlInsert.java]: fixed

**Review by jsternberg — APPROVED** [review]
